### PR TITLE
Lodash: add _.camelCase string array support 

### PR DESCRIPTION
**Pull request by mouadennaciri**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://lodash.com/docs/4.17.15#camelCase>

### CODE REVIEW

File: types/lodash/lodash-tests.ts [hunk 1]
```diff
@@ -5854,17 +5854,17 @@ fp.now(); // $ExpectType number
         _.transform<number, AbcObject>(dictionary, iterator); // $ExpectType Dictionary<AbcObject>
         _.transform<number, AbcObject>(dictionary, iterator, accumulator); // $ExpectType Dictionary<AbcObject>
         _(dictionary).transform(); // $ExpectType LoDashImplicitWrapper<Dictionary<any>>
-        _(dictionary).transform<number, AbcObject>(iterator); // $ExpectType LoDashImplicitWrapper<Dictionary<AbcObject>>
-        _(dictionary).transform<number, AbcObject>(iterator, accumulator); // $ExpectType LoDashImplicitWrapper<Dictionary<AbcObject>>
+        _(dictionary).transform(iterator); // $ExpectType LoDashImplicitWrapper<Dictionary<AbcObject>>
+        _(dictionary).transform(iterator, accumulator); // $ExpectType LoDashImplicitWrapper<Dictionary<AbcObject>>
```
Comment (mouadennaciri) [inline types/lodash/lodash-tests.ts]: Fix a strange error when I run `npm run lint lodash`
```
The 'use-default-type-parameter' rule threw an error in '.../DefinitelyTyped/types/lodash/lodash-tests.ts':
TypeError: Cannot read property 'default' of undefined
```

**Comment by thorn0** [comment]
As per the docs the parameter is `string`. While it indeed works for `string[]` too because Lodash calls `.toString()` on the argument, generally we don't add undocumented signatures to the type definitions.

**Review by thorn0 — CHANGES_REQUESTED** [review]
As per the docs the parameter is string. While it indeed works for string[] too because Lodash calls .toString() on the argument, generally we don't add undocumented signatures to the type definitions.

So I don't think this PR can be merged.

**Comment by mouadennaciri** [comment]
> As per the docs the parameter is string. While it indeed works for string[] too because Lodash calls .toString() on the argument, generally we don't add undocumented signatures to the type definitions.
> 
> So I don't think this PR can be merged.

@thorn0 According to the docs _.camelCase can take a string array.

> Arguments
> [string=''] (string): The string to convert.

[_.parseInt](https://lodash.com/docs/4.17.15#parseInt) is for example a utility which takes only string.

> Arguments
> string (string): The string to convert.

**Comment by thorn0** [comment]
It's not a string array, this notation means that it's an optional parameter.

**Comment by mouadennaciri** [comment]
> It's not a string array, this notation means that it's an optional parameter.

@thorn0 You're right, ok I close the PR.

**Pull request closed without merging**